devel/githubContrib.go

```diff
@@ -0,0 +1,253 @@
+
+// This script goes from org -> repos (skipping forks) -> contributors -> user
+// -> guess/normalize the company and count contribs
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+// Check checks for non nil error and dies upon error
+func Check(err error, msg string) {
+	if err != nil {
+		log.Fatal(msg, err)
+	}
+}
+
+// getToken gets auth token from the env
+func getToken() string {
+	// fmt.Println("in GetToken")
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		log.Fatal("Need to have GITHUB_TOKEN set in the env")
+	}
+	return token
+}
+
+var gAuthToken = getToken() // Any other way to get a static ?
+
+// GitHubAPIURL returns the full v3 rest api for a given path
+func GitHubAPIURL(path string) string {
+	return "https://api.github.com/" + path
+}
+
+// NewGhRequest makes a GitHub request (with Accept and Authorization headers)
+func NewGhRequest(url string) *http.Request {
+	// fmt.Println("in GhRequest")
+	req, err := http.NewRequest("GET", url, nil)
+	Check(err, "Unable to make request")
+	req.Header.Add("Accept", "application/vnd.github.v3+json")
+	req.Header.Add("Authorization", "token "+gAuthToken)
+	req.Header.Add("User-Agent", "githubContribExtractor")
+	return req
+}
+
+// GetBodyForURL gets the body or dies/abort on any error
+func GetBodyForURL(url string) []byte {
+	req := NewGhRequest(url)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	Check(err, "Unable to send request")
+	body, err := ioutil.ReadAll(resp.Body)
+	Check(err, "Unable to read response")
+	succ := resp.StatusCode
+	log.Printf("Got %d : %s for %s", succ, resp.Status, url)
+	if succ != http.StatusOK {
+		os.Exit(1)
+	}
+	return body
+}
+
+// PrettyPrintJSON outputs indented version of the Json body (debug only)
+func PrettyPrintJSON(body []byte) {
+	var out bytes.Buffer
+	err := json.Indent(&out, body, "", "  ")
+	Check(err, "Unable to Indent json")
+	_, err = os.Stdout.Write(out.Bytes())
+	Check(err, "Unable to output json")
+}
+
+// Repo is what we use from github rest api v3 listing repositories per org
+type Repo struct {
+	ID              int64  `json:"id"`
+	Name            string `json:"name"`
+	FullName        string `json:"full_name"`
+	ContributorsURL string `json:"contributors_url"`
+	IsFork          bool   `json:"fork"`
+}
+
+// UserC is what we care about from what we get from the ContributorsURL
+type UserC struct {
+	Login         string `json:"login"`
+	ID            int64  `json:"id"`
+	Contributions int64  `json:"contributions"`
+}
+
+// UserData is for the json we get from the /users/:username API call
+type UserData struct {
+	Name    string `json:"name"` // full name
+	Company string `json:"company"`
+	Email   string `json:"email"`
+}
+
+var fromEmailCount = 0 // global variable ftl (or ftw)
+
+// RemoveFromEnd returns a string without the suffix if found
+func RemoveFromEnd(s string, what string) string {
```

Comment:
why not just `strings.TrimSuffix()`?  The logic is exactly the same but in fewer lines. No need to reinvent.